slmsuite/misc/files.py

```diff
@@ -193,7 +193,9 @@ def recurse(group):
                 if decode_bytes:
                     if isinstance(data_, bytes):
                         data_ = bytes.decode(data_)
-                    elif isinstance(data_, np.ndarray) and isinstance(data_[0], bytes):
+                    elif np.isscalar(data_):
```

Comment:
Required? The first check of line 197 should take care of this.

Reply:
`data_` could be an int or float. I think I added that specifically to prevent an error.